test/kitchen/test/integration/win-agent-with-cws-option/rspec_datadog/win-agent-with-cws-option_spec.rb

```diff
@@ -0,0 +1,131 @@
+            expect(File).to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.inf")
+        else
+            expect(is_windows_service_installed("datadog-security-agent")).to be_falsey
+            expect(is_windows_service_installed("ddprocmon")).to be_falsey
+
+            program_files = safe_program_files
+            expect(File).not_to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.cat")
+            expect(File).not_to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.sys")
+            expect(File).not_to exist("#{program_files}\\DataDog\\Datadog Agent\\bin\\agent\\ddprocmon.inf")
+
+        end
+    end
+    it 'does not have the driver running on install' do
+        ## verify that the driver is not started yet
+        expect(is_service_running?("ddprocmon")).to be_falsey
+    end
+
+end
+  
+shared_examples_for 'a Windows Agent with CWS running' do
+    # We retrieve the value defined in kitchen.yml because there is no simple way
+    # to set env variables on the target machine or via parameters in Kitchen/Busser
+    # See https://github.com/test-kitchen/test-kitchen/issues/662 for reference
+    let(:expect_cws_installed) {
+        parse_dna().fetch('dd-agent-rspec').fetch('cws_included')
+    }
+    it 'can start security agent' do
+        if expect_cws_installed
+            sa_conf_path = ""
+            sp_conf_path = ""
+            if os != :windows
+                sa_conf_path = "/etc/datadog-agent/security-agent.yaml"
+                sp_conf_path = "/etc/datadog-agent/system-probe.yaml"
+            else
+                sa_conf_path = "#{ENV['ProgramData']}\\Datadog\\security-agent.yaml"
+                sp_conf_path = "#{ENV['ProgramData']}\\Datadog\\system-probe.yaml"
+            end
+            f = File.read(sa_conf_path)
+            confYaml = YAML.load(f)
+            if !confYaml
+                confYaml = {}
+            end
+            if !confYaml.key("runtime_security_config")
+                confYaml["runtime_security_config"] = {}
+            end
+            confYaml["runtime_security_config"]["enabled"] = true
+            File.write(sa_conf_path, confYaml.to_yaml)
+        
+            spf = File.read(sp_conf_path)
+            spconfYaml = YAML.load(spf)
+            if !spconfYaml
+                spconfYaml = {}
+            end
+            if !spconfYaml.key("runtime_security_config")
+                spconfYaml["runtime_security_config"] = {}
+            end
+            spconfYaml["runtime_security_config"]["enabled"] = true
+            File.write(sp_conf_path, spconfYaml.to_yaml)
+        
+            expect(is_service_running?("datadog-system-probe")).to be_falsey
+            expect(is_service_running?("datadog-security-agent")).to be_falsey
+            #restart "datadog-agent"
+            stop "datadog-agent"
+            sleep 10
```

Comment:
Can we replace the `sleep` with a periodic check as to if the the service stopped and then only proceed if it stopped within an acceptable time period? Arbitrary sleeps can contribute to flaky tests, especially when it comes to Windows service shutdown and startup.

I think this is important because these are new Windows Services and there very well could be a bug in the start/stop/restart behavior.

For example (pseudo code) could be have something like:
```
def wait_for(action, checkIntervalInSeconds, maxTimeoutInSeconds)

if(!wait_for(service_stopped("datadog-agent", 1, 10)):
  fail the test, the action didn't happen within the specified maxTimeout period
```

This I think will bound the time we expect the services to startup / shutdown and not proceed with the test if/when that action failed.

I recognize that the `maxTimeout` is similar to an arbitrary sleep, but I think the benefit here is that we won't proceed if the timeout is exceeded.

Reply:
With the above in mind, if the subsequent call to `start` would fail / throw an error if the service was still "started", then we might not need the above. But if it silently continues, then the `expect` calls below could find that the service is running, but it is running because it never stopped instead of knowing that it stopped, was started, and is now running again.

Reply:
so I'm not sure if I can do that in ruby.  I suppose I could. But as you point out
- waitfor(arbitrary time) is functionally the same as (sleep for) arbitrary time
- and either way, if the service isn't stopped at the end of the arbitrary time period, the test will fail and not proceed.

So I'm missing the improvement.

Reply:
The improvement is that we can control the max timeout we're willing to accept on shutdown and then if that is exceeded, we know that it was because it happened as a direct result of not occurring soon enough.

> and either way, if the service isn't stopped at the end of the arbitrary time period, the test will fail and not proceed.

Is that the behavior right now? As written the test is:
```
stop
sleep
start
sleep
check_is_still_running
```
When we check that it is still running, how do we know that it is running after a successful stop / start vs. the stop failing and it ran the entire time?

Reply:
The stop() and start() functions achieve the desired state or fail.

Reply:
To close the thread, the sleeps that were unneeded were removed and the remaining `sleep 30` gives enough time for agent to get into steady state before shutting it down again.